src/mon/OSDMonitor.cc

```diff
@@ -1126,31 +1126,43 @@ void OSDMonitor::encode_pending(MonitorDBStore::TransactionRef t)
     }
   }
 
-  // tell me about it
-  for (auto i = pending_inc.new_state.begin();
-       i != pending_inc.new_state.end();
-       ++i) {
-    int s = i->second ? i->second : CEPH_OSD_UP;
-    if (s & CEPH_OSD_UP)
-      dout(2) << " osd." << i->first << " DOWN" << dendl;
-    if (s & CEPH_OSD_EXISTS)
-      dout(2) << " osd." << i->first << " DNE" << dendl;
-  }
-  for (map<int32_t,entity_addr_t>::iterator i = pending_inc.new_up_client.begin();
-       i != pending_inc.new_up_client.end();
-       ++i) {
-    //FIXME: insert cluster addresses too
-    dout(2) << " osd." << i->first << " UP " << i->second << dendl;
-  }
-  for (map<int32_t,uint32_t>::iterator i = pending_inc.new_weight.begin();
-       i != pending_inc.new_weight.end();
-       ++i) {
-    if (i->second == CEPH_OSD_OUT) {
-      dout(2) << " osd." << i->first << " OUT" << dendl;
-    } else if (i->second == CEPH_OSD_IN) {
-      dout(2) << " osd." << i->first << " IN" << dendl;
-    } else {
-      dout(2) << " osd." << i->first << " WEIGHT " << hex << i->second << dec << dendl;
+  {
+    OSDMap tmp;
+    // tell me about it
+    for (auto i = pending_inc.new_state.begin();
+      i != pending_inc.new_state.end(); ++i) {
+      int s = i->second ? i->second : CEPH_OSD_UP;
+      int32_t osd = i->first;
+      const entity_addr_t cluster = tmp.get_cluster_addr(osd);
+      if (s & CEPH_OSD_UP) 
+        dout(2) << "cluster " << cluster << " "
```

Comment:
Calling this `cluster` may be confusing - I would instead go with `cluster_addr`. Also, I would suggest moving the cluster addr to after `osd`. E.g.,

`dout(2) << "osd." << osd << " cluster_addr " << cluster_addr << " ..." << dendl;`

Reply:
Thanks for your suggestion!